src/update_agent/actor.rs

```diff
@@ -138,17 +144,48 @@ impl UpdateAgent {
         std::time::Duration::from_secs(secs.saturating_add(jitter))
     }
 
+    /// Log at INFO level how many and which deployments will be excluded from being
+    /// future update targets.
+    fn log_excluded_depls(deployments: Result<BTreeSet<Release>, ()>, actor: &UpdateAgent) {
```

Comment:
This argument feels a bit weird, I think you should take a `&BTreeSet<Release>` here and wrap the calling site for this with a `if let Ok(ref depls) = foo() { log_excluded_depls(depls) }`.

Reply:
ah yes, this helper function's argument is kind of weird, thanks!